website/www/site/content/en/documentation/sdks/python-multi-language-pipelines.md

```diff
@@ -0,0 +1,213 @@
+```
+
+`ExternalTransform` takes three parameters:
+
+* The URN for the cross-language transform
+* The payload, either as a byte string or a [PayloadBuilder](https://beam.apache.org/releases/pydoc/current/apache_beam.transforms.external.html#apache_beam.transforms.external.PayloadBuilder)
+* An expansion service
+
+The URN is simply a unique Beam identifier for the transform, and the expansion service has already been discussed. The PayloadBuilder is a new concept, discussed next.
+
+> **NOTE**: To ensure that your URN doesn't run into confilcts with URNs from other transforms, follow the URN conventions described at [Selecting a URN for Cross-language Transforms](/documentation/programming-guide/#1314-selecting-a-urn-for-cross-language-transforms).
+
+## Provide a payload builder
+
+The Python pipeline example above provides an [ImplicitSchemaPayloadBuilder](https://beam.apache.org/releases/pydoc/current/apache_beam.transforms.external.html#apache_beam.transforms.external.ImplicitSchemaPayloadBuilder) as the second argument to `ExternalTransform`. The `ImplicitSchemaPayloadBuilder` builds a payload that generates a schema from the provided values. In this case, the provided values are contained in the following key-value pair: `{'prefix': 'java:'}`. The `JavaPrefix` transform expects a `prefix` argument, and the payload builder passes in the string `java:`, which will be prepended to each input element.
+
+In general, payload builders help build the payload for the transform in the expansion request. If you provide a Beam schema to the payload builder, the builder uses it to perform a type/field mapping. Instead of the `ImplicitSchemaPayloadBuilder`, you could use a [NamedTupleBasedPayloadBuilder](https://beam.apache.org/releases/pydoc/current/apache_beam.transforms.external.html#apache_beam.transforms.external.NamedTupleBasedPayloadBuilder), which builds a payload based on a named tuple schema, or an [AnnotationBasedPayloadBuilder](https://beam.apache.org/releases/pydoc/current/apache_beam.transforms.external.html#apache_beam.transforms.external.AnnotationBasedPayloadBuilder), which builds a schema based on type annotations. For a complete list of available payload builders, see the [transforms.external API reference](https://beam.apache.org/releases/pydoc/current/apache_beam.transforms.external.html).
+
+## Use standard element types
+
+At a multi-language boundary, you have to use element types that all the Beam SDKs understand. These are types represented by the [Beam standard coders](https://github.com/apache/beam/blob/42e1ae8f8d07fb9c6fde14bd688c5a4b763d9d6e/model/pipeline/src/main/proto/beam_runner_api.proto#L784):
+
+* `BYTES`
+* `STRING_UTF8`
+* `KV`
+* `BOOL`
+* `VARINT`
+* `DOUBLE`
+* `ITERABLE`
+* `TIMER`
+* `WINDOWED_VALUE`
+* `ROW`
+
+For arbitrary structured types (for example, an arbitrary Java object), use `ROW` (`PCollection<Row>`). You may have to develop a new Java composite transform that produces a `PCollection<Row>`.
+
```

Comment:
Also add:

"You many use SDK specific coders within a composite cross-language transform, if such coders do not get used by PCollections that are consumed by the other SDKs."

Reply:
Done